### DISCUSSION TITLE
I use another openai service(not directly from openai) how to to solve it?

### DISCUSSION THREAD

**Discussion by OriLiMu** [danielmiessler/Fabric#961] [Q&A]
I use a openai online service not directly from openai. Like the poe, which can provide api-key. 
I fill the env variable .
OPENAI_API_KEY=sk-xxx
OPENAI_API_BASE_URL=https://api.bltapi.top/v1
But I get error when I run fabric command
cat test.md | fabric --pattern  extract_wisdom error, status code: 401, message: Incorrect API key provided: sk-Cs9yP***************************************C222. You can find your API key at https://platform.openai.com/account/api-keys.

Strange thing is when I run commnd fabric --setup and fill the api-key and base url for the first time. The models can be loaded correctly. 
But When I rerun the `fabric --setup` command The models load failed. And show the same errors above. 

Why this happens and how to solve it?

**Comment by fonteijne** [comment]
There is a fix deployed for a bug in the `.env` file (https://github.com/danielmiessler/fabric/issues/997)

Could you retry? Maybe it fixed your issue too.
